File: Sesion 7/Marco teorico/10diferentes_herencias/10.py
class Base:
    def __init__(self):
        self.publico = 10
        self.protegido = 20
        self.__privado = 30

class DerivadaPublica(Base):
    def mostrar(self):
        print("Herencia pública:", self.publico, ",", self.protegido)

class DerivadaProtegida(Base):
    def mostrar(self):
        print("Herencia protegida:", self.publico, ",", self.protegido)

class DerivadaPrivada(Base):
    def mostrar(self):
        print("Herencia privada:", self.publico, ",", self.protegido)


d1 = DerivadaPublica()
d1.mostrar()  
d2 = DerivadaProtegida()
d3 = DerivadaPrivada()

File: Sesion 7/Marco teorico/10diferentes_herencias/10.java
import java.util.*;

class Base {
    public int publico = 10;
    protected int protegido = 20;
    private int privado = 30;
}

class DerivadaPublica extends Base {
    public void mostrar() {
        System.out.println("Herencia pública: " + publico + ", " + protegido);
    }
}

class DerivadaProtegida extends Base {
    public void mostrar() {
        System.out.println("Herencia protegida: " + publico + ", " + protegido);
    }
}

class DerivadaPrivada extends Base {
    public void mostrar() {
        System.out.println("Herencia privada: " + publico + ", " + protegido);
    }
}

public class Main {
    public static void main(String[] args) {
        DerivadaPublica d1 = new DerivadaPublica();
        d1.mostrar(); 

        DerivadaProtegida d2 = new DerivadaProtegida();

        DerivadaPrivada d3 = new DerivadaPrivada();
    }
}
